qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,373 @@
+
+        for _ in range(self._max_iter):
+            self._check_generation(fun)
+            self._truncation()
+            self._update_vector()
+
+            best_mae_local = min(self._evaluations)
+
+            history.append(best_mae_local)
+            best_ind_local = np.where(self._evaluations == best_mae_local)[0][0]
+            best_ind_local = self._generation[best_ind_local]
+
+            # update the best values ever
+            if best_mae_local < self._best_cost_global:
+                self._best_cost_global = best_mae_local
+                self._best_ind_global = best_ind_local
+                not_better_count = 0
+
+            else:
+                not_better_count += 1
+                if not_better_count == self._dead_iter:
+                    break
+
+            self._new_generation()
+
+        result.x = self._best_ind_global
+        result.fun = self._best_cost_global
+        result.nfev = len(history) * self._size_gen
+
+        return result
+
+    @property
+    def size_gen(self) -> int:
+        """Returns the size of the generations (number of individuals per generation)"""
+        return self._size_gen
+
+    @size_gen.setter
+    def size_gen(self, value: int):
+        """
+        Sets the size of the generations of the algorithm.
+
+        Args:
+            value: Size of the generations (number of individuals per generation).
+
+        Raises:
+            ValueError: If `value` is lower than 1.
+        """
+        if value <= 0:
+            raise ValueError("The size of the generation should be greater than 0.")
+        self._size_gen = value
+
+    @property
+    def max_iter(self) -> int:
+        """Returns the maximum number of iterations"""
+        return self._max_iter
+
+    @max_iter.setter
+    def max_iter(self, value: int):
+        """
+        Sets the maximum number of iterations of the algorithm.
+
+        Args:
+            value: Maximum number of iterations of the algorithm.
+
+        Raises:
+            ValueError: If `value` is lower than 1.
+        """
+        if value <= 0:
+            raise ValueError("The maximum number of iterations should be greater than 0.")
+
+        self._max_iter = value
+
+    @property
+    def alpha(self) -> float:
+        """Returns the alpha parameter value (percentage of population selected to update
+        probabilistic model)"""
+        return self._alpha
+
+    @alpha.setter
+    def alpha(self, value: float):
+        """
+        Sets the alpha parameter (percentage of individuals selected to update the probabilistic
+        model)
+
+        Args:
+            value: Percentage [0,1] of generation selected to update the probabilistic model.
+
+        Raises:
+            ValueError: If `value` is lower than 0 or greater than 1.
+        """
+        if (value < 0) or (value > 1):
+            raise ValueError(
+                "Alpha represents a percentage and should be greater than 0 but lower " "than 1"
```

Comment:
```suggestion
                f"alpha must be in the range 0 to 1, value given was {value}"
```
This will include whatever the value that was passed in the error message too as it can useful information for users to see what it was when things failed.

Reply:
So we allow 0 as a value here - thats valid? It seems to be used to as a factor to multiply to form a truncate is I read the code `Percentage [0, 1] of the population to be selected as elite selection.` So 0 selects none and 1 selects all of them, right?

Reply:
I have applied the docstring modification. Also, the alpha parameter should be greater than 0, so i have hanged to <= instead of <.